src/HBaseClient.java

```diff
@@ -1038,21 +1038,31 @@ public Scanner newScanner(final String table) {
    * deferred {@link Scanner.Response} if HBase 0.95 and up.
    */
   Deferred<Object> openScanner(final Scanner scanner) {
-    num_scanners_opened.increment();
-    return sendRpcToRegion(scanner.getOpenRequest()).addCallbacks(
-      scanner_opened,
-      new Callback<Object, Object>() {
-        public Object call(final Object error) {
-          // Don't let the scanner think it's opened on this region.
-          scanner.invalidate();
-          return error;  // Let the error propagate.
-        }
-        public String toString() {
-          return "openScanner errback";
-        }
-      });
+    return openScanner(scanner,scanner.getOpenRequest());
   }
 
+  /**
+   * Overloaded openScanner to enable direct passing in of the HBaseRpc 
+   * @param scanner The scanner to open.
+   * 
+   */
+  Deferred<Object> openScanner(final Scanner scanner, final HBaseRpc open_req){
+    num_scanners_opened.increment();
+    HBaseRpc req = open_req;
+    return sendRpcToRegion(req).addCallbacks(
+          scanner_opened,
+          new Callback<Object, Object>() {
+            public Object call(final Object error) {
+              // Don't let the scanner think it's opened on this region.
+              scanner.invalidate();
+              return error;  // Let the error propagate.
+            }
+            public String toString() {
+              return "openReverseScanner errback";
```

Comment:
So this openScanner is not just for reverse scan, right? Then why use "openReverseScanner here"?